pkg/store/storepb/rpc.proto

```diff
@@ -122,6 +122,12 @@ message SeriesRequest {
   // shard_info is used by the querier to request a specific
   // shard of blocks instead of entire blocks.
   ShardInfo shard_info = 13;
+
+  // maximum_string_slots is the number of maximum strings
+  // that the receiver can send. It is safe to ignore this by
+  // the receiver if it does not intend on sending a symbol table
+  // via hints later on.
+  uint64 maximum_string_slots = 14;
```

Comment:
Why do we need to set a maximum limit here?

Reply:
Because symbol references are uint64 and because a Querier might be connected to N different StoreAPIs, each sending references it means that it could use up to `math.Uint64Max/N` references. Without this limit, a Querier might not be able to proxy all of those references it receives to another querier.

Reply:
Got it, makes sense 👍